packages/dev/docs/pages/react-spectrum/testing.mdx

```diff
@@ -0,0 +1,158 @@
+can be.
+
+## Testing semantics
+
+Many testing libraries expect you to query for elements in the DOM tree. For example, you might have a test 
+that renders a login page, finds the username and password fields, and simulates filling them out and 
+submitting the form.
+
+Querying for these elements can be done in many ways. It's common to use CSS selectors for this, 
+but this can be problematic if you rely on implementation details of the component in these queries.
+For example, if you query by a CSS class name that may change in a future update of React Spectrum, your
+test will be brittle.
+
+This also applies if you query for internal elements within a React Spectrum component. The DOM structure
+that React Spectrum renders should be considered a black box, and it may change at any time, so you should
+avoid relying on it.
+
+The correct way to query for React Spectrum components and their internals is by semantics. React Spectrum
+components implement [ARIA patterns](https://www.w3.org/TR/wai-aria-practices-1.2/). ARIA is a W3C standard
+that specifies the semantics for many UI components. This is used to expose them to assistive technology
+like screen readers, but can also be used in tests to operate the application programmatically. These semantics
+are much less likely to change over time, and while other DOM nodes may be added or removed, and CSS classes
+changed, the semantics will stay stable.
+
+The main attribute to look for when querying is the [role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques).
+This attribute represents the type of element a DOM node represents, e.g. a button, list option, or tab.
+This is similar to the HTML element type (e.g. `<button>` or `<option>`), but since many React Spectrum
+components are implemented with `<div>` for styling purposes, the `role` attribute is used instead to
+communicate the element's semantics.
+
+### React Testing Library
+
+[React Testing Library](https://testing-library.com/docs/react-testing-library/intro) is great because it
+enforces that you write tests using semantics instead of implementation details. We use React Testing Library
+to test React Spectrum itself, and it's quite easy to [query](https://testing-library.com/docs/dom-testing-library/api-queries)
+elements by role, text, label, etc.
+
+```tsx
+import {render} from '@testing-library/react';
+
+let tree = render(<MyComponent />);
+let option = tree.getByRole('button');
+```
+
+### Enzyme
+
+In Enzyme, you can [use CSS selectors](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/find.html) to find
+elements. You can query by role using an attribute selector:
+
+```tsx
+import {mount} from 'enzyme';
+
+let wrapper = mount(<MyComponent />);
+let option = wrapper.find('[role="option"]');
+```
+
+## Test ids
+
+Querying by semantics is great, but what if you have many buttons on a page? How do you find the specific button
```

Comment:
Querying by semantics covers many scenarios, however what if you need to query for many buttons...